File: app/src/main/java/com/cartenz/cartenztaxcore/feature/login/LoginContract.java
package com.cartenz.cartenztaxcore.feature.login;


import com.cartenz.core.BasePresenter;
import com.cartenz.core.BaseView;

public interface LoginContract {
    interface View extends BaseView<PresenterInterface> {
        void loginResult(String error);

    }

    interface PresenterInterface extends BasePresenter<View> {

        void callLogin(String username, String password);
    }
}

File: kotlinapp/src/main/java/com/cartenz/kotlinapp/feature/login/LoginContract.kt
package com.cartenz.kotlinapp.feature.login

import com.cartenz.kotlin_core.CartenzBasePresenter
import com.cartenz.kotlin_core.CartenzBaseView

interface LoginContract {
    interface ViewCartenz : CartenzBaseView<PresenterInterfaceCartenz> {
        fun loginResult(error: String)
    }

    interface PresenterInterfaceCartenz : CartenzBasePresenter<ViewCartenz> {
        fun callLogin(username: String, password: String)
    }

}
